docs/current_docs/manuals/developer/functions.mdx

```diff
@@ -338,19 +338,116 @@ The result will look like this:
+│   ├── ...
+│   └── src
+│       └── main
+├── index.html
+├── public
+│   └── ...
+├── src
+│   └── ...
+```
+
+Here are two Dagger Functions with default contexts for their directory and file arguments:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+The default context is set by adding a `defaultPath` pragma on the corresponding Dagger Function argument.
+
+```go file=./snippets/functions/arguments-context-dir-file/go/main.go
+```
+</TabItem>
+<TabItem value="Python">
+
+The default context is set by adding a `DefaultPath` annotation on the corresponding Dagger Function argument.
+
+```python file=./snippets/functions/arguments-context-dir-file/python/main.py
+```
+</TabItem>
+<TabItem value="TypeScript">
+
+The default context is set by adding an `@argument` decorator with a `defaultPath` parameter on the corresponding Dagger Function argument.
+
+```typescript file=./snippets/functions/arguments-context-dir-file/typescript/index.ts
+```
+</TabItem>
+</Tabs>
+
+The following table describes how the arguments are resolved for differing values of the default context and differing directory types.
+
+For a Git repository:
+
+| Default context ('defaultPath`) | Dagger Function invocation | `source` resolves to |
+| --- | --- | --- |
+| `/` | `dagger call -m my-module read-dir` | `/` |
+| `/src` | `dagger call -m my-module read-dir` | `/src` |
+| `.` | `dagger call -m my-module read-dir` | `/my-module` |
+| `..` | `dagger call -m my-module read-dir` | `/` |
+| `/README.md` | `dagger call -m my-module read-file` | `/README.md` |
+| `./README.md` | `dagger call -m my-module read-file` | Error: file does not exist |
+| `../README.md` | `dagger call -m my-module read-file` | `/README.md` |
+
+For a non-Git directory:
+
+| Default context ('defaultPath`) | Dagger Function invocation | `source` resolves to |
+| --- | --- | --- |
+| `/` | `dagger call -m my-module read-dir` | `/my-module` |
+| `/src` | `dagger call -m my-module read-dir` | `/my-module/src` |
+| `.` | `dagger call -m my-module read-dir` | `/my-module` |
+| `..` | `dagger call -m my-module read-dir` | Error: escaping default context directory root is not allowed |
+| `/README.md` | `dagger call -m my-module read-file` | Error: file does not exist |
+| `./README.md` | `dagger call -m my-module read-file` | Error: file does not exist |
+| `../README.md` | `dagger call -m my-module read-file` | Error: escaping default context directory root is not allowed |
+
+If a value is explicitly passed to the Dagger Function argument, it always overrides the default context.
+
+| Default context ('defaultPath`) | Dagger Function invocation | `source` resolves to |
+| --- | --- | --- |
+| `/` | `dagger call -m my-module read-dir --source=public` | `/public` |
+| `/src` | `dagger call -m my-module read-dir --source=public` | `/public` |
+| `.` | `dagger call -m my-module read-dir --source=public` | `/public` |
+| `..` | `dagger call -m my-module read-dir --source=public` | `/public` |
+
+:::tip
```

Comment:
We should add an example in a follow up wih the execution output so show how ignore works.